validator/client/grpc-api/grpc_validator_client.go

```diff
@@ -117,8 +118,16 @@ func (c *grpcValidatorClient) WaitForActivation(ctx context.Context, in *ethpb.V
 }
 
 // Deprecated: Do not use.
-func (c *grpcValidatorClient) WaitForChainStart(ctx context.Context, in *empty.Empty) (ethpb.BeaconNodeValidator_WaitForChainStartClient, error) {
-	return c.beaconNodeValidatorClient.WaitForChainStart(ctx, in)
+func (c *grpcValidatorClient) WaitForChainStart(ctx context.Context, in *empty.Empty) (*ethpb.ChainStartResponse, error) {
+	stream, err := c.beaconNodeValidatorClient.WaitForChainStart(ctx, in)
+	if err != nil {
+		return nil, errors.Wrap(
+			iface.ErrConnectionIssue,
+			errors.Wrap(err, "could not setup beacon chain ChainStart streaming client").Error(),
+		)
```

Comment:
This is going to mess up the error stack. Why is it wrapped so strange?

Reply:
I didn't add this code, I just moved it from [validator/client/validator.go](https://github.com/prysmaticlabs/prysm/pull/11654/files#diff-f0ef7fb9371dbf4c04023b673da6d03a5c008cad259f0a3acc26603cfbb33159) so I'm not sure if it's formatted this way. I didn't change the error message because I don't want to change the existing gRPC behavior.

Reply:
OK, fair enough